### PR TITLE
Allow editing ways that are pending upload

### LINKED ISSUE #285: Editing ways that are pending upload
Right now, we don't allow selecting or editing ways that are pending upload. With #282, this is partly fixed but we need to handle cases like:
* [x] adding a new node to a way that's pending upload
* [x] deleting a node from the way
* [x] moving
* [x] merging
* [ ] preset changes

cc @batpad

### PR DESCRIPTION
**Pull request by geohacker**
Fixes #285

### CODE REVIEW

**Comment by geohacker** [comment]
Merging nodes doesn't work currently. I can look into that tomorrow.

**Comment by sethvincent** [comment]
Did some testing of this and I think it's working as expected. It'll be easier to tell once we merge in latest from edit-ways.

File: app/reducers/wayEditingHistory.js [hunk 1]
```diff
@@ -18,10 +18,30 @@ function wayEditingHistory (state = createDefaultState(), action) {
...
+      let movedNodes
+      let deletedNodes
+      let mergedNodes
+      let modifiedSharedWays
       console.log('way being edited', way)
+      if (action.wayEditingHistory) {
+        // editing a way that's pending upload
+
+        const wayEditingHistory = _cloneDeep(action.wayEditingHistory)
+        addedNodes = wayEditingHistory.addedNodes
```
Comment (batpad) [inline app/reducers/wayEditingHistory.js:31]: can this be simplified to `{addedNodes, movedNodes, deletedNodes, mergedNodes, modifiedSharedWays} = wayEditingHistory` ?

**Comment by sethvincent** [comment]
I've done some testing of adding, moving, deleting, and merging nodes in ways that are pending upload and i think this is looking good.

**Pull request merged**